modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/APIService.java

```diff
@@ -260,4 +262,89 @@ public Response importAPI(@Multipart("file") InputStream uploadedInputStream, @Q
+     * @param httpHeaders HTTP headers for authentication mechanism
+     * @return A JSON response indicating the status of the update
+     */
+    @PUT
+    @Path("/{apiID}")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateAPI(@Multipart("file") InputStream uploadedInputStream, @PathParam("apiID") String apiID,
+                              @QueryParam("preserveProvider") String defaultProviderStatus, @Context HttpHeaders httpHeaders) {
+        boolean isTenantFlowStarted = false;
+        boolean isProviderPreserved = true;
+
+        //Check if the URL parameter value is specified, otherwise the default value "true" is used
+        if (APIImportExportConstants.STATUS_FALSE.equalsIgnoreCase(defaultProviderStatus)) {
+            isProviderPreserved = false;
+        }
+
+        try {
+            Response authorizationResponse = AuthenticatorUtil.authorizeUser(httpHeaders);
+
+            if (Response.Status.OK.getStatusCode() != authorizationResponse.getStatus()) {
+                return authorizationResponse;
+            }
+
+            AuthenticationContext authenticationContext = (AuthenticationContext) authorizationResponse.getEntity();
+            String tenantDomain = MultitenantUtils.getTenantDomain(authenticationContext.getUsername());
+            String currentUser = authenticationContext.getDomainAwareUsername();
+
+            if (!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+                currentUser = currentUser + "@" + tenantDomain;
+            }
+            APIImportUtil.initializeProvider(currentUser);
+
+            //Temporary directory is used to create the required folders
+            String currentDirectory = System.getProperty(APIImportExportConstants.TEMP_DIR);
+            String createdFolders = File.separator +
+                    RandomStringUtils.randomAlphanumeric(APIImportExportConstants.TEMP_FILENAME_LENGTH) +
+                    File.separator;
+            File importFolder = new File(currentDirectory + createdFolders);
+            boolean folderCreateStatus = importFolder.mkdirs();
+
+            //API import process starts only if the required folder is created successfully
+            if (folderCreateStatus) {
+                String uploadFileName = APIImportExportConstants.UPLOAD_FILE_NAME;
+                String absolutePath = currentDirectory + createdFolders;
+                APIImportUtil.transferFile(uploadedInputStream, uploadFileName, absolutePath);
+
+                String extractedFolderName;
+                try {
+                    extractedFolderName = APIImportUtil.extractArchive(
+                            new File(absolutePath + uploadFileName), absolutePath);
+                } catch (APIImportException e) {
+                    return Response.status(Status.BAD_REQUEST).entity(e.getMessage()).build();
+                }
+
+                if (tenantDomain != null &&
```

Comment:
Do we need to check for a null here?

Reply:
I think it can be removed :)
Checked the code and the method returns carbon.super if nothing found. So it is not gonna be null

Reply:
What I'm trying to say here is, even if it is null it does not going to affect the equals method as we are avoiding the NPE by using it as follows,
`MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)`

Hope you got it :)